### PR TITLE
layers: Update SPIRV-Tools for GPU-AV improvements

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
The descriptor indexing and buffer OOB instrumentation code
has been streamlined and combined. Update GPU-AV to the new interface.

These configuration options are removed, since it is no longer possible
to enable these 2 features independently:

```
khronos_validation.gpuav_descriptor_indexing
khronos_validation.gpuav_buffer_oob
```

They are replaced by the following option:

```
khronos_validation.gpuav_descriptor_checks
```

NOTE: This change requires updating SPIRV-Tools

### CODE REVIEW

File: layers/gpu_validation/gpu_validation.cpp [hunk 1]
```diff
@@ -961,8 +961,11 @@ bool GpuAssisted::InstrumentShader(const vvl::span<const uint32_t> &input, std::
     opt_options.set_validator_options(val_options);
     Optimizer optimizer(target_env);
     optimizer.SetMessageConsumer(gpu_console_message_consumer);
-    optimizer.RegisterPass(CreateInstBindlessCheckPass(desc_set_bind_index, unique_shader_module_id, descriptor_indexing,
-                                                       descriptor_indexing, buffer_oob_enabled, buffer_oob_enabled));
+    // Descriptor and OOB validation are now tightly coupled
+    if (descriptor_indexing || buffer_oob_enabled) {
```
Comment (jeremyg-lunarg) [inline layers/gpu_validation/gpu_validation.cpp:965]: @Tony-LunarG and @christophe-lunarg  This change makes the `gpuav_buffer_oob `and `gpuav_descriptor_indexing ` options in the json  file be redundant. It is no longer possible to enable one without enabling the other.  It seems like there are 3 choices for what I can do here:

1. keep both options and this block of code
2. keep only one of the existing options and remove the other from the VVL code and json
3. make up a new option with a more useful name and remove the 2 existing options.


What do you think?

Comment (Tony-LunarG) [reply layers/gpu_validation/gpu_validation.cpp:965]: I think a new option and better description makes the most sense.  Also we should note the removal of the other options in the release notes for the next SDK.

Comment (jeremyg-lunarg) [reply layers/gpu_validation/gpu_validation.cpp:965]: Now I just need to come up with a good name. THAT shouldn't be hard 😜

File: layers/VkLayer_khronos_validation.json.in [hunk 2]
```diff
@@ -848,29 +848,9 @@
...
-                                                ]
-                                            }
-                                        },
-                                        {
-                                            "key": "gpuav_buffer_oob",
-                                            "label": "Check Out of Bounds",
-                                            "description": "Enable buffer out of bounds checking",
+                                            "key": "gpuav_descriptor_checks",
+                                            "label": "Descriptor and OOB Checks",
+                                            "description": "Enable descriptor and buffer out of bounds checking",
```
Comment (jeremyg-lunarg) [inline layers/VkLayer_khronos_validation.json.in:853]: @Tony-LunarG and @christophe-lunarg how about this? Open to other suggestions for the name of the option or the text here.

**Comment by y-novikov** [comment]
Is it possible to expedite this?
Rolling VVL into ANGLE is blocked on this after https://github.com/KhronosGroup/SPIRV-Tools/pull/5274 landed.
https://chromium-review.googlesource.com/c/angle/angle/+/4639488

**Comment by jeremyg-lunarg** [comment]
@y-novikov I'm working on this today. Unfortunately we haven't come up with a nice way to deal with the codependency between GPU-AV and SPIRV-Tools. I could move this code and the parts of SPIRV-Tools into VVL's repo, but copying code makes @dneto0 have a sad.

File: layers/gpu_shaders/gpu_shaders_constants.h [hunk 3]
```diff
@@ -15,17 +15,19 @@
 // limitations under the License.
 // Values used between the GLSL shaders and the GPU-AV logic
 
-#pragma once
+// NOTE: glslang includes this header and it doesn't support #pragma once
```
Comment (juan-lunarg) [inline layers/gpu_shaders/gpu_shaders_constants.h:18]: glslang includes a header from the vulkan validation layers?

Comment (jeremyg-lunarg) [reply layers/gpu_shaders/gpu_shaders_constants.h:18]: More accurately, the shaders in layers/gpu_shaders include it and glslang doesn't support pragma once

Comment (juan-lunarg) [reply layers/gpu_shaders/gpu_shaders_constants.h:18]: That makes much more sense. Could you rephrase this?

File: layers/gpu_validation/gpu_validation.cpp [hunk 4]
```diff
@@ -1031,42 +1029,45 @@ bool GenerateValidationMessage(const uint32_t *debug_record, std::string &msg, s
...
-                strm << "(set = " << debug_record[kInstBindlessBuffOOBOutDescSet] << ", binding = " << debug_record[kInstBindlessBuffOOBOutDescBinding]
-                     << ") Descriptor index " << debug_record[kInstBindlessBuffOOBOutDescIndex]
-                     << " access out of bounds. Descriptor size is " << debug_record[kInstBindlessBuffOOBOutBuffSize]
-                     << " texels and highest texel accessed was " << debug_record[kInstBindlessBuffOOBOutBuffOff];
-                if (debug_record[kInstValidationOutError] == kInstErrorBuffOOBUniformTexel)
-                    vuid_msg = vuid.uniform_access_oob;
-                else
-                    vuid_msg = vuid.storage_access_oob;
+            oob_access = true;
+            switch (binding_state->descriptor_class) {
```
Comment (Tony-LunarG) [inline layers/gpu_validation/gpu_validation.cpp:1047]: Nice that the instrumentation doesn't have to know about descriptor type anymore

File: layers/gpu_validation/gv_descriptor_sets.cpp [hunk 5]
```diff
@@ -172,7 +172,11 @@ void gpuav_state::DescriptorSet::SetBindingState(uint32_t *data, uint32_t index,
                     data[index++] = vvl::kU32Max;
                 } else {
                     auto buffer_view_state = desc.GetBufferViewState();
-                    data[index++] = static_cast<uint32_t>(buffer_view_state->buffer_state->createInfo.size);
+                    VkDeviceSize view_size = buffer_view_state->create_info.range;
+                    if (view_size == VK_WHOLE_SIZE) {
```
Comment (Tony-LunarG) [inline layers/gpu_validation/gv_descriptor_sets.cpp:176]: Ugh, how long has this been broken?  A test using VK_WHOLE_SIZE might be good (but not in this PR)

**Review by Tony-LunarG — APPROVED** [review]
Very nice!!  LGTM

File: layers/gpu_validation/gv_descriptor_sets.cpp (hunk 5 above)
Comment (jeremyg-lunarg) [reply layers/gpu_validation/gv_descriptor_sets.cpp:176]: a long time I think
